### PR TITLE
LPD-35277

### PR DESCRIPTION
**Pull request by vngoliveira**
[LPD-35277](https://liferay.atlassian.net/browse/LPD-35277)

[LPD-35277]: https://liferay.atlassian.net/browse/LPD-35277?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/info/collection/provider/test/ManyToManyObjectRelationshipInfoCollectionProviderTest.java [hunk 1]
```diff
@@ -159,6 +162,58 @@ public void testManyToManyObjectRelationshipRelatedInfoCollectionProvider()
 		_assertRelatedInfoCollectionProvider(
 			_objectDefinition2, objectDefinition2ObjectEntry,
 			objectDefinition1ObjectEntry1, objectDefinition1ObjectEntry2);
+
+		Map<String, ServiceRegistration<?>> serviceRegistrationMap =
+			_objectRelationshipLocalService.getServiceRegistrations();
```
Comment (marcelabc) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/info/collection/provider/test/ManyToManyObjectRelationshipInfoCollectionProviderTest.java:167]: Rename to `serviceRegistrations`  please

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/info/collection/provider/test/ManyToManyObjectRelationshipInfoCollectionProviderTest.java [hunk 2]
```diff
@@ -159,6 +162,58 @@ public void testManyToManyObjectRelationshipRelatedInfoCollectionProvider()
...
+
+		ServiceRegistration<?>
+			objectRelationshipDefinition1ServiceRegistration =
```
Comment (marcelabc) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/info/collection/provider/test/ManyToManyObjectRelationshipInfoCollectionProviderTest.java:170]: Rename to `ObjectRelationshipServiceRegistration`

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/info/collection/provider/test/ManyToManyObjectRelationshipInfoCollectionProviderTest.java [hunk 3]
```diff
@@ -159,6 +162,58 @@ public void testManyToManyObjectRelationshipRelatedInfoCollectionProvider()
...
+					StringBundler.concat(
+						_objectRelationship.getCompanyId(), StringPool.POUND,
+						_objectRelationship.getObjectRelationshipId()));
+
+		objectRelationshipDefinition1ServiceRegistration.unregister();
+
+		serviceRegistrationMap.remove(
+			StringBundler.concat(
+				_objectRelationship.getCompanyId(), StringPool.POUND,
+				_objectRelationship.getObjectRelationshipId()));
```
Comment (marcelabc) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/info/collection/provider/test/ManyToManyObjectRelationshipInfoCollectionProviderTest.java:181]: Will the list be updated if we execute `_objectRelationshipLocalService.getServiceRegistrations()`?
If yes > It is better to execute the get method and assert that this ServiceRegistration is not in that map.

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/info/collection/provider/test/ManyToManyObjectRelationshipInfoCollectionProviderTest.java [hunk 4]
```diff
@@ -159,6 +162,58 @@ public void testManyToManyObjectRelationshipRelatedInfoCollectionProvider()
...
+
+		ObjectRelationship objectRelationshipDefinition2 =
```
Comment (marcelabc) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/info/collection/provider/test/ManyToManyObjectRelationshipInfoCollectionProviderTest.java:183]: Rename to `reverseObjectRelationship`

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/info/collection/provider/test/ManyToManyObjectRelationshipInfoCollectionProviderTest.java [hunk 5]
```diff
@@ -159,6 +162,58 @@ public void testManyToManyObjectRelationshipRelatedInfoCollectionProvider()
...
+						StringPool.POUND,
+						objectRelationshipDefinition2.
+							getObjectRelationshipId()));
+
+		objectRelationshipDefinition2ServiceRegistration.unregister();
+
+		serviceRegistrationMap.remove(
+			StringBundler.concat(
+				objectRelationshipDefinition2.getCompanyId(), StringPool.POUND,
+				objectRelationshipDefinition2.getObjectRelationshipId()));
```
Comment (marcelabc) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/info/collection/provider/test/ManyToManyObjectRelationshipInfoCollectionProviderTest.java:202]: Will the list be updated if we execute _objectRelationshipLocalService.getServiceRegistrations()?
If yes > It is better to execute the get method and assert that this ServiceRegistration is not in that map.

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectDisplayPageTemplates.testcase [hunk 6]
```diff
@@ -43,4 +43,31 @@ definition {
...
+
+		DisplayPageTemplatesAdmin.openDisplayPagesAdmin(siteURLKey = "guest");
+
+		LexiconEntry.gotoAdd();
+
+		LexiconCard.clickCard(card = "Blank");
+
+		Click(locator1 = "VisualizeObjectDisplayPage#OBJECT_CONTENT_TYPE");
+	}
+
```
Comment (marcelabc) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectDisplayPageTemplates.testcase:72]: Can you remove this code from the buildService commit? This type of commit can only contain generated code.

**Review by marcelabc — CHANGES_REQUESTED** [review]
Hi 
I finished my review. Let me know if you have questions :smile:

**Comment by vngoliveira** [comment]
ci:test:sf

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/info/collection/provider/test/ManyToManyObjectRelationshipInfoCollectionProviderTest.java [hunk 7]
```diff
@@ -159,6 +162,62 @@ public void testManyToManyObjectRelationshipRelatedInfoCollectionProvider()
...
+
+		ServiceRegistration<?> objectRelationshipServiceRegistration =
+			serviceRegistrations.get(serviceRegistrationKey);
+
+		objectRelationshipServiceRegistration.unregister();
+
+		serviceRegistrations.remove(serviceRegistrationKey);
+
+		serviceRegistrations =
+			_objectRelationshipLocalService.getServiceRegistrations();
```
Comment (marcelabc) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/info/collection/provider/test/ManyToManyObjectRelationshipInfoCollectionProviderTest.java:181]: Why do we need to call the remove method if we execute the get method in the following line?
I think we can remove the line calling the remove method.

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/info/collection/provider/test/ManyToManyObjectRelationshipInfoCollectionProviderTest.java [hunk 8]
```diff
@@ -159,6 +162,62 @@ public void testManyToManyObjectRelationshipRelatedInfoCollectionProvider()
...
+		serviceRegistrationKey = StringBundler.concat(
+			reverseObjectRelationship.getCompanyId(), StringPool.POUND,
+			reverseObjectRelationship.getObjectRelationshipId());
+
+		objectRelationshipServiceRegistration = serviceRegistrations.get(
+			serviceRegistrationKey);
+
+		objectRelationshipServiceRegistration.unregister();
+
+		serviceRegistrations.remove(serviceRegistrationKey);
```
Comment (marcelabc) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/info/collection/provider/test/ManyToManyObjectRelationshipInfoCollectionProviderTest.java:200]: same observation. Please remove this line.

**Review by marcelabc — CHANGES_REQUESTED** [review]
Hi 
I finished my review. Let me know if you have questions :smile:

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/info/collection/provider/test/ManyToManyObjectRelationshipInfoCollectionProviderTest.java [hunk 9]
```diff
@@ -159,6 +162,62 @@ public void testManyToManyObjectRelationshipRelatedInfoCollectionProvider()
...
+		String serviceRegistrationKey = StringBundler.concat(
+			_objectRelationship.getCompanyId(), StringPool.POUND,
+			_objectRelationship.getObjectRelationshipId());
+
+		ServiceRegistration<?> objectRelationshipServiceRegistration =
+			serviceRegistrations.get(serviceRegistrationKey);
+
+		objectRelationshipServiceRegistration.unregister();
+
+		serviceRegistrations.remove(serviceRegistrationKey);
```
Comment (marcelabc) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/info/collection/provider/test/ManyToManyObjectRelationshipInfoCollectionProviderTest.java:178]: Please verify if it is possible to:

- assert that there is no related entry when asserting the _objectDefinition1:
```
_assertRelatedInfoCollectionProvider(
	_objectDefinition1, objectDefinition1ObjectEntry1,
	objectDefinition2ObjectEntry);
_assertRelatedInfoCollectionProvider(
	_objectDefinition1, objectDefinition1ObjectEntry2,
	objectDefinition2ObjectEntry);
```
- assert that there is related entries when asserting the _objectDefinition2:
```
_assertRelatedInfoCollectionProvider(
	_objectDefinition2, objectDefinition2ObjectEntry,
	objectDefinition1ObjectEntry1, objectDefinition1ObjectEntry2);
```

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/info/collection/provider/test/ManyToManyObjectRelationshipInfoCollectionProviderTest.java (hunk 8 above)
Comment (marcelabc) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/info/collection/provider/test/ManyToManyObjectRelationshipInfoCollectionProviderTest.java:200]: Please verify if it is possible to:

- assert that there is no related entry when asserting the _objectDefinition1:
```
_assertRelatedInfoCollectionProvider(
	_objectDefinition1, objectDefinition1ObjectEntry1,
	objectDefinition2ObjectEntry);
_assertRelatedInfoCollectionProvider(
	_objectDefinition1, objectDefinition1ObjectEntry2,
	objectDefinition2ObjectEntry);
```
- assert that there are no related entries when asserting the _objectDefinition2:
```
_assertRelatedInfoCollectionProvider(
	_objectDefinition2, objectDefinition2ObjectEntry,
	objectDefinition1ObjectEntry1, objectDefinition1ObjectEntry2);
```

**Comment by vngoliveira** [comment]
ci:test:sf

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/info/collection/provider/test/ManyToManyObjectRelationshipInfoCollectionProviderTest.java [hunk 10]
```diff
@@ -159,6 +162,88 @@ public void testManyToManyObjectRelationshipRelatedInfoCollectionProvider()
...
+				RelatedInfoItemCollectionProvider.class,
+				StringBundler.concat(
+					ObjectDefinition.class.getName(), StringPool.POUND,
+					_objectDefinition2.getObjectDefinitionId())));
+
+		serviceRegistrations =
+			_objectRelationshipLocalService.getServiceRegistrations();
+
+		Assert.assertFalse(
+			serviceRegistrations.containsKey(serviceRegistrations));
```
Comment (marcelabc) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/info/collection/provider/test/ManyToManyObjectRelationshipInfoCollectionProviderTest.java:197]: I think this code is unnecessary. Can you remove it, please?

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/info/collection/provider/test/ManyToManyObjectRelationshipInfoCollectionProviderTest.java [hunk 11]
```diff
@@ -159,6 +162,88 @@ public void testManyToManyObjectRelationshipRelatedInfoCollectionProvider()
...
```
Comment (marcelabc) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/info/collection/provider/test/ManyToManyObjectRelationshipInfoCollectionProviderTest.java:232]: I think this code is unnecessary. Can you remove it, please?

**Review by marcelabc — CHANGES_REQUESTED** [review]
Hi 
I finished my review. Let me know if you have questions :smile:

**Comment by vngoliveira** [comment]
ci:test:sf

**Review by marcelabc — APPROVED** [review]
LGTM 🎉

**Comment by marcelabc** [comment]
ci:test:sf

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by marcelabc** [comment]
ci:test:object

**Comment by marcelabc** [comment]
ci:report:6967539

**Comment by marcelabc** [comment]
ci:report:6974268

**Comment by marcelabc** [comment]
There are no unique failures.

**Comment by marcelabc** [comment]
ci:forward:force

**Pull request closed without merging**